src/memusage.h

```diff
@@ -25,6 +25,64 @@ namespace memusage
 /** Compute the total memory used by allocating alloc bytes. */
 static size_t MallocUsage(size_t alloc);
 
+/** Wrapping allocator that accounts accurately.
+ *
+ * To use accounting, the AccountAllocator(size_t&) constructor must be used.
+ * Any containers using such an allocator, and containers moved constructed or
+ * assigned from such containers will then increment/decrement size_t when
+ * allocating/deallocating memory.
+ */
+template<typename T>
+class AccountingAllocator : public std::allocator<T>
+{
+    //! Pointer to accounting variable, if any.
+    size_t* m_allocated;
+
+    template<typename U> friend class AccountingAllocator;
+    typedef std::allocator<T> base;
+
+public:
+    //! Default constructor constructs a non-accounting allocator.
+    AccountingAllocator() noexcept : m_allocated(nullptr) {}
+
+    /** Construct an allocator that increments/decrements 'allocated' on allocate/free.
+     *
+     * In a multithreaded environment, the variable needs to be protected by the same
```

Comment:
65fc229fabbc7889baca9c50ee35507543a6682e
```suggestion
     * In a multithreaded environment, the accounting variable needs to be protected by the same
```